spec/grammars/count_grammar_parser_spec.rb

```diff
@@ -121,6 +121,25 @@ module RapSheetParser
         expect(count.code_section.section.text_value).to eq '484-487 (A)'
       end
 
+      it 'handles stray information when charge is in the comments' do
+        text = <<~TEXT
+          SEE COMMENT FOR CHARGE
+          TOC:N
+          Page 22 of 26
+          *DISPO:CONVICTED
+          CONV STATUS:MISDEMEANOR
+          SEN: IMP SEN SS, 003 YEARS PROBATION, 020 DAYS JAIL, FINE
+          COM: CNT 01 CHRG-594(A)-(B) (2) (A) PC
+          DCN:10000000000000000007
+        TEXT
+
+        count = described_class.new.parse(text)
+        expect(count.charge_line.text_value).to eq('SEE COMMENT FOR CHARGE')
+        expect(count.disposition.disposition_type).to be_a CountGrammar::Convicted
+        expect(count.code_section.code.text_value).to eq 'PC'
+        expect(count.code_section.section.text_value).to eq '594(A) - (B)(2)(A)'
```

Comment:
This line currently fails with:
```
       expected: "594(A) - (B)(2)(A)"
            got: "594(A)-(B) (2) (A)"
```
Do we expect the spaces in `594(A)-(B) (2) (A) PC` (in the source text) to be modified by the parser? I.e. spaces added around the `-`, and removed between the parens

Reply:
I'm a little confused... it looks like the text you wrote in the expectation is different from the text in the RAP sheet. If you change the expectation to match the RAP sheet code section, it works. Is there a reason you thought the spaces should be modified?
```suggestion
        expect(count.code_section.section.text_value).to eq '594(A)-(B) (2) (A)'
```

Reply:
> Is there a reason you thought the spaces should be modified?

Ah, I didn't explain that clearly: the expected value here is taken from `expected_values.json` (line 372) for this test case:
```
        {
          "code_section": "PC 594(A) - (B)(2)(A)",
          "severity": "M",
...
```

Reply:
Ahh, I see. The difference is that the unit test here is a unit test of the CountGrammar. It's taking the raw text value that the parser found for `count.code_section.section`, which we would expect to match exactly with the RAP sheet text.

The `expected_values.json` file is part of an integration test that is testing the grammar, builder, and model objects together. If you look in [count_builder.rb:52](https://github.com/codeforamerica/rap_sheet_parser/blob/16f6f40db925b218e1b604d1ea3d2e3fdb8e77a4/lib/rap_sheet_parser/builders/count_builder.rb#L52), you'll see that the builder is doing some cleanup on the text, deleting spaces and replacing commas with periods (probably because we saw that issue with the OCR, and code sections generally don't contain commas).

So for that unit test, it's appropriate to change the expectation as I suggested above.

Reply:
> If you look in count_builder.rb:52, you'll see that the builder is doing some cleanup on the text

@lkogler As a quick follow-up point, even with the additional cleanup in `count_builder`, the integration test fails:
```
-   [{"code_section"=>"PC 594(A) - (B)(2)(A)",
-     "sentence"=>"imp sen ss, ...y probation, ...d jail, fine",
+   [{"code_section"=>"PC 594(A)-(B)(2)(A)",
+     "sentence"=>"...y probation, ...d jail, imp sen ss, fine",
```
I.e. the spaces in `(B) (2) (A)` are indeed removed (to give `(B)(2)(A)`), but `)-(` is **not** turned to `) - (`, as `expected_values.json` expects.

Should `expected_values` be changed for that, or is this "missing" logic in the `count_builder` cleanup?

Reply:
Probably we just made a mistake in `expected_values.json` 🙃

Reply:
Ah, that's the easiest fix, then ;-) Thanks for clarifying!